django/db/backends/base/schema.py

```diff
@@ -431,9 +433,7 @@ def _column_generated_sql(self, field):
         """Return the SQL to use in a GENERATED ALWAYS clause."""
         expression_sql, params = field.generated_sql(self.connection)
         persistency_sql = "STORED" if field.db_persist else "VIRTUAL"
-        if params:
-            expression_sql = expression_sql % tuple(self.quote_value(p) for p in params)
-        return f"GENERATED ALWAYS AS ({expression_sql}) {persistency_sql}"
+        return f"GENERATED ALWAYS AS ({expression_sql}) {persistency_sql}", params
```

Comment:
The crux of the issue here is that
1. This part was never `%` escaped (`%` -> `%%`) which mean that things were working properly during model creation unless the model included a database default which resulted in `table_sql` returning `params` and then interpolation kicked (see the `params or None` in `create_model`). As for `add_field` it systematically crashed because a list was always passed to `execute(params)` which triggered interpolation which is what is what was reported on ticket-35336.
2. Postgres simply doesn't support parametrized generated expressions, it will crash with a very clear message about it.